crates/matrix-sdk-sled/src/state_store.rs

```diff
@@ -585,6 +613,26 @@ impl SledStore {
 
         self.save_room_timeline(changes).await?;
 
+        // user state
+        let ret: Result<(), TransactionError<SledStoreError>> = (&self.session, &self.account_data)
```

Comment:
We now have 3 transactions? Doesn't this fit into the previous one like presence did?

Reply:
I tried splitting them around semantics after not being able to put more entries into the existing one. Might be able to coble that into two, but not quite the right split. But I guess either is bad as it leaves incomplete state, if any one fails... so my choice was to at least have them semantically grouped...

Reply:
I guess as long as the sync token is persisted last we're fineish, no matter how many transactions we have.